src/libs/Trie/index.js

```diff
@@ -0,0 +1,86 @@
+import _ from 'underscore';
+import TrieNode from './TrieNode';
+
+class Trie {
+    constructor() {
+        this.root = new TrieNode();
+    }
+
+    /**
+    * Add a word to the Trie
+    * @param {String} word
+    * @param {Object} [metaData]
+    * @param {TrieNode} node
+    * @param {String} [containChar]
+    * @returns {void}
+    */
+    add(word, metaData, node = this.root, containChar) {
+        if (word.length === 0 && !containChar) {
+            throw new Error('Cannot insert empty word into Trie :', word);
+        }
+        if (word.length === 0) {
+            node.setCompleteWord();
+            node.setMetaData(metaData);
+            return;
+        } if (!node.children[word[0]]) {
+            // eslint-disable-next-line no-param-reassign
+            node.children[word[0]] = new TrieNode();
+            return this.add(word.substring(1), metaData, node.children[word[0]], word.charAt(0));
+        }
+        return this.add(word.substring(1), metaData, node.children[word[0]], word.charAt(0));
+    }
+
+    /**
+    * Check if the word is exist in the Trie.
+    * @param {String} word
+    * @returns {Object}
+    */
+    isWord(word) {
+        let node = this.root;
+        while (word.length > 1) {
+            if (!node.children[word[0]]) {
+                return null;
+            }
+            node = node.children[word[0]];
+            // eslint-disable-next-line no-param-reassign
+            word = word.substring(1);
```

Comment:
Can we address this @Karim-30, and handle it properly?

Are we fine with this @stitesExpensify?

Reply:
No, I agree with you. We should fix the actual problem instead of disabling the linter in almost all situations.

Reply:
Done.